### DISCUSSION TITLE
Generate multiple runs while incrementing op configs

### DISCUSSION THREAD

**Discussion by SgtGilly** [dagster-io/dagster#15729] [Q&A] [answered]
Hello,

I am trying to figure out how I can programmatically generate multiple runs of a job in which I am increment a value of an op config within that job.

For example say I have an op called **A** inside a job called **B**. Op **A** is passed a config at runtime and one of the values of that config is an int with a default value of 2. I want to increment that default value by 2 up to 10 and run that job for each new value. So I essentially want to run job **B** five times incrementing op **A's** configurable int value by 2 in each run. I want to also have tags that show me what that config value is for each run. 

Is there a way to achieve this in Dagster? 

An initial approach I thought of (which I think is ultimately cumbersome) is to leverage the run_status_sensor to generate new RunRequest with an update run_config field for the same job over and over until the condition of op_config_value > 10 is met. Additionally, I can then add the new config value to the tags field of the RunRequest.

Please let me know if this question makes sense. Thank you!!

**Comment by yuhan** [comment, marked as answer]
I'd recommend using a [sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors) to kick off N runs of the same job, where you can increment the value and specify the tag on the RunRequest, such as:
```python
from dagster import sensor, RunRequest

@sensor(job=my_job)
def my_sensor():
    for i in range(2, 11):
        yield RunRequest(
            run_key=str(i),
            tags={"value": str(i)},
        )
```

In addition to that, if you'd like to parameterize the starting value, you can [use resources in sensors](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#using-resources-in-sensors):
```python
from dagster import sensor, RunRequest, ConfigurableResource

class MyValue(ConfigurableResource):
    start: int
    end: int

@sensor(job=my_job)
def my_sensor(my_value: MyValue):
    for i in range(my_value.start, my_value.end):
        yield RunRequest(
            run_key=str(i),
            tags={"value": str(i)},
        )
```
Then, you can provide the values through `Definitions`:
```python
defs = Definitions(
    jobs=[my_job],
    sensors=[my_sensor],
    resources={"my_value": MyValue(start=2, end=11)},
)
```
